tests/helpers.py

```diff
@@ -103,13 +158,25 @@ def check_subgraph_escrow_account(sender, receiver, totalAmountThawing, balance)
     else:
         raise Exception(f"Request failed with status code: {resp.status_code}")
 
+def signer_data_check(data, isThawing, authorized):
+    if data[0]["isAuthorized"] != authorized:
+        return True
+    if isThawing:
+        if int(data[0]["thawEndTimestamp"]) <= 0:
+            return True
+    else:
+        if int(data[0]["thawEndTimestamp"]) != 0:
+            return True
+    return False
```

Comment:
This function should be refactored. Many returns and nested ifs. I cannot make sure it's doing what it's supposed to do.
Make sure that it is very easy to read.